concrete/controllers/single_page/login.php

```diff
@@ -388,6 +391,20 @@ public function logout($token = false)
         }
     }
 
+    /**
+     * @param $token
+     * @return RedirectResponse
+     */
+    public function do_logout($token = false)
+    {
+        if ($this->app->make('token')->validate('do_logout', $token)) {
+            $u = new User();
```

Comment:
We shouldn't use the User constructor to get the currently logged in user any longer. Use `$this->app->make(User::class)` which does the same thing today but makes it easier to make this a singleton in the future.